src/bun.js/node/node_os.zig

```diff
@@ -54,11 +54,147 @@ pub const Os = struct {
         return JSC.ZigString.init(Global.arch_name).withEncoding().toValue(globalThis);
     }
 
+    const CPU = struct {
+        model: JSC.ZigString = JSC.ZigString.init("unknown"),
+        speed: u64 = 0,
+        times: struct {
+            user: u64 = 0,
+            nice: u64 = 0,
+            sys: u64 = 0,
+            idle: u64 = 0,
+            irq: u64 = 0,
+        } = .{}
+    };
+
     pub fn cpus(globalThis: *JSC.JSGlobalObject, _: *JSC.CallFrame) callconv(.C) JSC.JSValue {
         JSC.markBinding(@src());
 
-        // TODO:
-        return JSC.JSArray.from(globalThis, &.{});
+        var cpu_buffer: [8192]CPU = undefined;
+        const cpus_or_error = if (comptime Environment.isLinux)
+                                cpusImplLinux(&cpu_buffer)
+                              else
+                                cpu_buffer[0..0];  // unsupported platform -> empty array
+
+        if (cpus_or_error) |list| {
+            // Convert the CPU list to a JS Array
+            const values = JSC.JSValue.createEmptyArray(globalThis, list.len);
+            for (list) |cpu, cpu_index| {
+                const obj = JSC.JSValue.createEmptyObject(globalThis, 3);
+                obj.put(globalThis, JSC.ZigString.static("model"), cpu.model.withEncoding().toValue(globalThis));
```

Comment:
Confusingly, `toValue` assumes if the pointer was not tagged as globally allocated then it is statically allocated but this is temporary memory so it will need to be `toValueGC` to ensure that the string is cloned

```suggestion
                obj.put(globalThis, JSC.ZigString.static("model"), cpu.model.withEncoding().toValueGC(globalThis));
```